nerfstudio/data/utils/nerfstudio_collate.py

```diff
@@ -163,11 +163,38 @@ def nerfstudio_collate(batch: Any, extra_mappings: Union[Dict[type, Callable], N
             op = torch.cat
 
         # Create metadata dictionary
-        metadata_keys = batch[0].metadata.keys()
-        assert all(
-            (cam.metadata.keys() == metadata_keys for cam in batch)
-        ), "All cameras must have the same metadata keys."
-        metadata = {key: op([cam.metadata[key] for cam in batch], dim=0) for key in metadata_keys}
+        if batch[0].metadata is not None:
+            metadata_keys = batch[0].metadata.keys()
+            assert all(
```

Comment:
Can we move this assert to the beginning of the function just like what we did for distortion_params?